summarization/summarizer_main.py

```diff
@@ -177,8 +191,8 @@ def _perform_title_first_matching(self):
 
                     sentence_media_pair_has_existing_stamp = True
 
-                if not sentence_media_pair_has_existing_stamp:
-                    unmatched_content.append(sentence_media_pair)
+            if not sentence_media_pair_has_existing_stamp:
+                unmatched_content.append(sentence_media_pair)
 
         self._assemble_and_add_stamp_pages_to_list(unmatched_content)
```

Comment:
Wrong indentation here caused the Stamp page to be added every for every iteration of the inner for loop.